exploration/0005-open-close-expressions.md

```diff
@@ -0,0 +1,152 @@
+# Open/Close Expressions
+
+Status: **Proposed**
+
+<details>
+	<summary>Metadata</summary>
+	<dl>
+		<dt>Contributors</dt>
+		<dd>@eemeli</dd>
+		<dd>@aphillips</dd>
+		<dt>First proposed</dt>
+		<dd>2023-09-05</dd>
+		<dt>Pull Request</dt>
+		<dd><a href="https://github.com/unicode-org/message-format-wg/pull/470">#470</a></dd>
+	</dl>
+</details>
+
+## Objective
+
+_What is this proposal trying to achieve?_
+
+Describe the use cases and requirements for _expressions_
+that enclose parts of a pattern
+and develop a design that satisfies these needs.
+
+## Background
+
+_What context is helpful to understand this proposal?_
+
+## Use-Cases
+
+- Representing markup, such as HTML, as expressions in a pattern
+  rather than as plain character sequences embedded in the _text_
+  of a pattern. This also allows parts of a markup sequence to be
+  programmable.
+
+**_Non-markup use cases to consider (which are not addressed by the design)_**
+
+- Represent templating language constructs so that they can pass through
+  the translation process and are not exposed to translation. For example,
+  Mustache templates.
+
+- Represent translation metadata, such as XLIFF placeholders, so that
+  portions of a message are "protected" or otherwise hinted in the
+  translation process.
+
+## Requirements
+
+Be able to indicate that some identified markup applies to
+a non-empty sequence of pattern parts (text, expressions).
+
+Markup spans may be nested,
+as in `<b>Bold and <i>also italic</i></b>`.
+
+Markup spans may be improperly nested,
+as in `<b>Bold <i>and</b> italic</i>`.
```

Comment:
Why is it a requirement to support this? I can see not caring at all, but rejecting designs outright that fail to accommodate improper nesting seems inappropriate.

Reply:
That's actually a fair point. I'd be happy to drop this. Anyone disagree?

```suggestion
```

Reply:
Your proposal focuses on HTML. I imagine that @mihnita might bring back such nesting for "attributed string" support. But I'd reserve that for when it is needed?